pkg/convert/testdata/programs/required_providers/pcl/main.pp

```diff
@@ -0,0 +1,12 @@
+package "different-name-null" {
+  baseProviderName = "different-name-null"
```

Comment:
This is why I said you need to look at .Source() at https://github.com/pulumi/pulumi-converter-terraform/pull/239/files#r1867179326. The base provider and package label here should be "null".

I'd also use a different provider than null, like http so that you don't have a package name conflict in this test.